pycbc/workflow/segment.py

```diff
@@ -1122,9 +1122,15 @@ def get_analyzable_segments(workflow, out_dir, tags=[]):
             sci_segs[ifo] -= cat_segs
             
         sci_segs[ifo].coalesce()
-        seg_ok_path = os.path.abspath(os.path.join(out_dir, '%s-SCIENCE-OK.xml' % ifo))
-        seg_files += [segments_to_file(sci_segs[ifo], seg_ok_path, 
-                                          "SCIENCE_OK", ifo=ifo)]  
+        seg_ok_path = os.path.abspath(os.path.join(out_dir,
+                                                   '%s-SCIENCE-OK.xml' % ifo))
+        curr_url = urlparse.urlunparse(['file', 'localhost', seg_ok_path,
+                          None, None, None])
+        curr_file = OutSegFile(ifo, 'SCIENCE_OK', workflow.analysis_time,
+                               curr_url, segment_list=sci_segs[ifo],
```

Comment:
I've already added it. I needed it to test the psd workflow html page, so
I'll include the commit in the pull request.

On Mon, Sep 28, 2015 at 3:34 PM, spxiwh <notifications@github.com> wrote:

> In pycbc/workflow/segment.py
> <https://github.com/ligo-cbc/pycbc/pull/381#discussion_r40551776>:
>
> > @@ -1122,9 +1122,15 @@ def get_analyzable_segments(workflow, out_dir, tags=[]):
> >              sci_segs[ifo] -= cat_segs
> >
> >          sci_segs[ifo].coalesce()
> > -        seg_ok_path = os.path.abspath(os.path.join(out_dir, '%s-SCIENCE-OK.xml' % ifo))
> > -        seg_files += [segments_to_file(sci_segs[ifo], seg_ok_path,
> > -                                          "SCIENCE_OK", ifo=ifo)]
> > +        seg_ok_path = os.path.abspath(os.path.join(out_dir,
> > +                                                   '%s-SCIENCE-OK.xml' % ifo))
> > +        curr_url = urlparse.urlunparse(['file', 'localhost', seg_ok_path,
> > +                          None, None, None])
> > +        curr_file = OutSegFile(ifo, 'SCIENCE_OK', workflow.analysis_time,
> > +                               curr_url, segment_list=sci_segs[ifo],
>
> Can you set it here:
>
> diff --git a/pycbc/workflow/core.py b/pycbc/workflow/core.py
> index 1177640..9b7e432 100644
> --- a/pycbc/workflow/core.py
> +++ b/pycbc/workflow/core.py
> @@ -1210,6 +1210,7 @@ class OutSegFile(File):
>          from pycbc.events import segments_to_file
>          segments_to_file(self.segmentList, self.storage_path,
>                               self.tagged_description,  ifo=self.ifo_string)
> +        self.PFN(self.storage_path, site='local')
>
>  def make_external_call(cmdList, out_dir=None, out_basename='external_call',
>                         shell=False, fail_on_error=True):
>
> I can test and commit this myself, unless it is easy for you to do it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ligo-cbc/pycbc/pull/381/files#r40551776>.
>